node-manager/src/node.rs

```diff
@@ -30,23 +36,73 @@ pub(crate) type PeerManager = LdkPeerManager<
     Arc<IgnoringMessageHandler>,
 >;
 
+type ChainMonitor = chainmonitor::ChainMonitor<
+    InMemorySigner,
+    Arc<dyn Filter + Send + Sync>,
+    Arc<MutinyWallet>,
+    Arc<MutinyWallet>,
+    Arc<MutinyLogger>,
+    Arc<MutinyNodePersister>,
+>;
+
 pub struct Node {
     pub uuid: String,
     pub pubkey: PublicKey,
     pub keys_manager: Arc<KeysManager>,
+    blockchain: Arc<EsploraBlockchain>,
 }
 
 impl Node {
-    pub(crate) fn new(node_index: NodeIndex, mnemonic: Mnemonic) -> Result<Self, MutinyError> {
+    pub(crate) fn new(
+        node_index: NodeIndex,
+        mnemonic: Mnemonic,
+        storage: MutinyBrowserStorage,
+        network: Network,
+    ) -> Result<Self, MutinyError> {
         info!("initialized a new node: {}", node_index.uuid);
 
-        let keys_manager = create_keys_manager(mnemonic, node_index.child_index);
+        // init the key manager
+        let keys_manager = Arc::new(create_keys_manager(
+            mnemonic.clone(),
+            node_index.child_index,
+        ));
         let pubkey = pubkey_from_keys_manager(&keys_manager);
 
+        // init the persister
+        let persister = Arc::new(MutinyNodePersister::new(
+            node_index.uuid.clone(),
+            storage.clone(),
+        ));
+
+        // init esplora
+        let blockchain = Arc::new(esplora_from_network(network));
+
+        // init wallet
+        let wallet = Arc::new(MutinyWallet::new(
+            mnemonic.clone(),
+            storage.clone(),
+            network,
+        ));
+
+        // init chain monitor
+        let chain_monitor: Arc<ChainMonitor> = Arc::new(chainmonitor::ChainMonitor::new(
+            None,
+            wallet.clone(),
+            Arc::new(MutinyLogger {}),
+            wallet,
+            persister.clone(),
+        ));
+
+        // read channelmonitor state from disk
+        let channelmonitors = persister
```

Comment:
@benthecarman this is where channel monitors is initialized, it's read in from the persister.